src/test/java/org/terasology/manualLabor/systems/ShearingSystemTest.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.manualLabor.systems;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.entity.internal.PojoEntityManager;
+import org.terasology.engine.rendering.assets.material.Material;
+import org.terasology.engine.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.engine.rendering.logic.SkeletalMeshComponent;
+import org.terasology.gestalt.assets.management.AssetManager;
+
+import java.util.Optional;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ShearingSystemTest {
+    ShearingSystem shearingSystem;
+    private EntityRef entity;
+
+    /**
+     * Initialize the shearing system and entity to be used while testing. The entity is supposed to mock the animal entity.
+     */
+    @BeforeEach
+    public void setUp() {
+        shearingSystem = new ShearingSystem();
+        entity = new PojoEntityManager().create();
+        SkeletalMeshComponent skeletalMeshComponent = new SkeletalMeshComponent();
+        entity.saveComponent(skeletalMeshComponent);
+    }
+
+    /**
+     * Check whether switchPrefab() changes mesh and material attributes of SkeletalMeshComponent correctly
+     */
+    @Test
+    public void testSwitchPrefabAssetChange() {
+        SkeletalMesh expectedMesh = mock(SkeletalMesh.class);
+        Material expectedMaterial = mock(Material.class);
+        createMockAssetManager(expectedMesh, expectedMaterial);
+
+        shearingSystem.switchPrefab(entity, "testMesh", "testMaterial");
+        SkeletalMeshComponent skeletalMeshComponent = entity.getComponent(SkeletalMeshComponent.class);
+
+        Assertions.assertEquals(skeletalMeshComponent.mesh, expectedMesh);
+        Assertions.assertEquals(skeletalMeshComponent.material, expectedMaterial);
+    }
+
+    /**
+     * Check whether switchPrefab() changes mesh and material attributes of SkeletalMeshComponent correctly when mesh is empty. Ideal
+     * behaviour in such a case would be to skip the change
```

Comment:
Similarly, if you are asserting that the value did _not_ change, have the test make it clear that the "after" value is the same as the "before" value.

Reply:
(I also suggest logging an error message when the method encounters an invalid condition, instead of continuing on silently, but you don't need to make that part of the test.)